miio/vacuum_tui.py

```diff
@@ -0,0 +1,95 @@
+try:
+    import curses
+except ImportError:
+    pass
+
+import enum
+from typing import Tuple, Type
+
+from .vacuum import Vacuum
+
+
+class Control(enum.Enum):
+
+    Quit = "q"
+    Forward = "l"
+    ForwardFast = "L"
+    Backward = "k"
+    BackwardFast = "K"
+    Left = "j"
+    LeftFast = "J"
+    Right = ";"
+    RightFast = ":"
+
+
+class VacuumTUI:
+    def __init__(self, vac: Vacuum):
+        self.vac = vac
+        self.rot = 0
+        self.rot_delta = 30
+        self.rot_min = Vacuum.MANUAL_ROTATION_MIN
+        self.rot_max = Vacuum.MANUAL_ROTATION_MAX
+        self.vel = 0.0
+        self.vel_delta = 0.1
+        self.vel_min = Vacuum.MANUAL_VELOCITY_MIN
+        self.vel_max = Vacuum.MANUAL_VELOCITY_MAX
+        self.dur = 10 * 1000
+
+    @classmethod
+    def run(cls: Type["VacuumTUI"], vac: Vacuum) -> None:
+        vac.manual_start()
+        try:
+            tui = cls(vac=vac)
+            curses.wrapper(tui.main)
+        finally:
+            vac.manual_stop()
+
+    def main(self, screen) -> None:
```

Comment:
```suggestion
    def loop(self, screen) -> None:
```
or `main_loop`?

Reply:
Renamed to `main_loop`.

Reply:
Ended up splitting the method into two:
- `loop` handles keys
- `main` prints usage and starts the loop